Sources/Profile-Modify.php

```diff
@@ -2892,7 +2892,7 @@ function profileValidateSignature(&$value)
 
 		$unparsed_signature = strtr(un_htmlspecialchars($value), array("\r" => '', '&#039' => '\''));
 		// Too long?
-		if (!empty($sig_limits[1]) && $smcFunc['strlen']($unparsed_signature) > $sig_limits[1])
+		if (!empty($sig_limits[1]) && (!empty($sig_bbc) ? ($smcFunc['strlen']($unparsed_signature) - $smcFunc['strlen']($sig_bbc)) : $smcFunc['strlen']($unparsed_signature)) > $sig_limits[1])
```

Comment:
I don't know if this is correct, because `$sig_bbc` appears to contain the list of disabled bbc tags (which is from a field in the admin cp). I think what you can do is check for the length of the signature with the bb codes stripped off -- `$unparsed_signature` contains the signature with the bb codes still.